src/chart-review/chart-review.component.tsx

```diff
@@ -1,87 +1,101 @@
-      name: "widget",
-      label: "Hello", //will be displayed on the nav bar
-      esModule: "@jj-widgets", //this module must be in the import map
-      path: "/widget" //this will be the path to the widget and will be appended to /patient/:patientUuid/chart
-    }
-  ] ;
-*/
+  const [routes, setRoutes] = React.useState([]);
+  const [navbarItems, setNavbarItems] = React.useState([]);
 
   const [selected, setSelected] = React.useState(getInitialTab());
-
-  React.useEffect(() => {
-    const externalWidgets: externalWidgetsType = {};
-    const promises = [];
-    const moduleMap = {};
-
-    config.widgetDefinitions.forEach(def => {
-      externalWidgets[def.name] = def;
-      //only import modules once
-      if (moduleMap[def.esModule] === undefined) {
-        promises.push(System.import(def.esModule));
-      }
-    });
-
-    Promise.all(promises).then(modules => {
-      //widgets is an array of objects, see type below
-      const widgets: widgetType[] = [];
-
-      //Promise.all returns an array of resolved modules.
-      // Place into an object with key = module name to make it easier to access in the below widget loadinng loop
-      modules.map(mod => {
-        moduleMap[mod.name] = mod;
-      });
-
-      //config.widgets is an array of widget names
-      config.widgets.map(widgetName => {
-        //First see if name exists in coreWidgets
-        if (coreWidgets[widgetName]) {
-          widgets.push(coreWidgets[widgetName]);
-        } else {
-          const widget = externalWidgets[widgetName];
-          let Component: FunctionComponent =
-            moduleMap[widget.esModule].widgets[widget.name];
-          widget.component = () => <Component />;
-
-          widgets.push(widget);
-        }
-      });
-
-      setWidgets(widgets);
-    });
-  }, [config, setWidgets]);
+  const [tabHistory, setTabHistory] = React.useState({});
 
   function getInitialTab() {
-    if (config == undefined || widgets.length === 0) {
+    if (config == undefined || navbarItems.length === 0) {
       return 0;
+      //need to rename from widget to something else
     } else if (widgetPath == undefined) {
       return config.defaultTabIndex;
     } else {
-      return widgets.findIndex(element => element.path === "/" + widgetPath);
+      return navbarItems.findIndex(
+        element => element.path === "/" + widgetPath
+      );
     }
   }
 
-  const [tabHistory, setTabHistory] = React.useState({});
+  function getCoreView(name: string): ViewType {
+    if (coreWidgets[name]) {
+      return coreWidgets[name];
+    }
+    if (coreDashboards[name]) {
+      return {
+        ...coreDashboards[name],
+        component: () => (
+          <Dashboard key={name} dashboardConfig={coreDashboards[name]} />
+        )
+      };
+    }
+    return;
+    //TODO: if(coreMultiDashboards[view])
+  }
+
+  function getExternalView(config, name: string): ViewType {
+    let i = config.widgetDefinitions.findIndex(item => item.name === name);
```

Comment:
this can be a little simplified :-)
```suggestion
    const widget = config.widgetDefinitions.find(widget => widget.name === name);
    const dashboard = config.dashboardDefinitions.find(dashboard => dashboard.name === name);
    if(widget){
         return { name, component: () => <Widget widgetConfig={widget} /> }
    } else if(dashboard) {
        return { name, component: () => <Dashboard dashboardConfig={dashboard}/>
    } else {
        return { name, component: null }
    }
    }
```

Reply:
Agreed. changes made!